language_server/src/request_manager.cpp

```diff
@@ -64,12 +70,8 @@ void request_manager::end_worker()
 
 bool hlasm_plugin::language_server::request_manager::is_running() 
 {
-    bool result = false;
-    {
-        std::unique_lock<std::mutex> lock(q_mtx_);
-        result = !requests_.empty();
-    }
-    return result;
+    std::lock_guard<std::mutex> lock(q_mtx_);
+    return !requests_.empty();
```

Comment:
Just a note, I did exactly this in the past and for some reason it was breaking the server... As long as it works now.

Reply:
Do you have any idea under what conditions it used to break? Because I am pretty sure these two implementations are equivalent.